odoo/addons/base/res/res_partner.py

```diff
@@ -324,7 +324,9 @@ def _check_parent_id(self):
     @api.multi
     def copy(self, default=None):
         self.ensure_one()
-        default = dict(default or {}, name=_('%s (copy)') % self.name)
+        chosen_name = default.get('name') if default else ''
+        new_name = chosen_name or _('%s (copy)') % self.name
```

Comment:
Why not just 

`new_name = default.get('name') if default else _('%s (copy)') % self.name`

Reply:
Because default.get('name') can be None or '' ?

Reply:
Or shorter again, just reverse the order so the provided name updates name but a little more confusing

`default = dict(name=_('%s (copy)') % self.name, **(default or {}))

Reply:
Fair point.  Just seemed strange the 2 lines to assign.

Reply:
Readability was also a motivation to have to lines.